packages/xo-server-sdn-controller/src/index.js

```
@@ -612,6 +638,38 @@ class SDNController extends EventEmitter {
     }
   }
 
+  async _addNetworkRule({ networkId, allow, direction, ipRange = '', port, protocol }) {
+    try {
+      const network = this._xo.getXapiObject(this._xo.getObject(networkId, 'network'))
+      assert(network.PIFs.length > 0, 'Network needs to be plugged to delete  rule')
+      const channel = this._getOrCreateOfChannel(network.PIFs[0].host)
+      await channel.addNetworkRule(network, allow, protocol, port, ipRange, direction)
+      const networkRules = network.other_config['xo:sdn-controller:of-rules']
+      const newNetworkRules = networkRules !== undefined ? JSON.parse(networkRules) : []
+      const stringRule = JSON.stringify({
+        allow,
+        protocol,
+        port,
+        ipRange,
+        direction,
+      })
+      if (!newNetworkRules.includes(stringRule)) {
+        newNetworkRules.push(stringRule)
+        await network.update_other_config('xo:sdn-controller:of-rules', JSON.stringify(newNetworkRules))
+      }
+    } catch (error) {
+      log.error('Error while adding Network OF rule', {
+        error,
+        networkId,
+        allow,
+        protocol,
+        port,
+        ipRange,
+        direction,
+      })
```

Is it intended that the error is ignored after being logged? (i.e. when calling this function, we have no way to tell if it ended successfully)
Same thing for `_deleteNetworkOfRule`

---

that was the case in the original addRule implementation